File: app/src/main/java/com/aquamorph/frcmanager/utils/Logging.java
package com.aquamorph.frcmanager.utils;

import android.util.Log;

/**
 * Parses date of type T.
 *
 * @author Christian Colglazier
 * @version 2/15/2018
 */

public class Logging {

    /**
     * debug() prints to debug at a tracing level
     * @param object
     * @param message
     * @param level
     */
    public static void debug(Object object, String message, int level) {
        if (Constants.TRACING_LEVEL >= level ) {
            Log.d(object.getClass().getSimpleName(), message);
        }
    }

    /**
     * error() prints to error at a tracing level
     * @param object
     * @param message
     * @param level
     */
    public static void error(Object object, String message, int level) {
        if (Constants.TRACING_LEVEL >= level ) {
            Log.e(object.getClass().getSimpleName(), message);
        }
    }
}

File: app/src/main/java/com/aquamorph/frcmanager/utils/Logging.kt
package com.aquamorph.frcmanager.utils

import android.util.Log

/**
 * Parses date of type T.
 *
 * @author Christian Colglazier
 * @version 3/30/2018
 */

object Logging {

    /**
     * debug() prints to debug at a tracing level
     * @param object
     * @param message
     * @param level
     */
    fun debug(`object`: Any, message: String, level: Int) {
        if (Constants.TRACING_LEVEL >= level) {
            Log.d(`object`.javaClass.simpleName, message)
        }
    }

    /**
     * info() prints to info at a tracing level
     * @param object
     * @param message
     * @param level
     */
    fun info(`object`: Any, message: String, level: Int) {
        if (Constants.TRACING_LEVEL >= level) {
            Log.i(`object`.javaClass.simpleName, message)
        }
    }

    /**
     * error() prints to error at a tracing level
     * @param object
     * @param message
     * @param level
     */
    fun error(`object`: Any, message: String, level: Int) {
        if (Constants.TRACING_LEVEL >= level) {
            Log.e(`object`.javaClass.simpleName, message)
        }
    }
}
